### PR TITLE
Added snippets and tsx setting

### PR DESCRIPTION
**Pull request by torn4dom4n**
Fixes broken snippets from https://github.com/atom/language-typescript/pull/2 and add tsx settings
/cc @damieng

### CODE REVIEW

**Comment by damieng** [comment]
Hi.

Are these from the upstream Microsoft package or from elsewhere?  I ask because they are tending to use JSON everywhere but these are in CSON format.

**Comment by torn4dom4n** [comment]
I used snippets from https://github.com/Microsoft/vscode/tree/master/extensions/typescript/snippets but I edited for Atom.

**Comment by damieng** [comment]
What changed were made?

I'm trying to avoid a scenario where we can't keep up to date with their
repo and undocumented changes will cause that.

Were the other files unchanged?

File: snippets/typescript.json [hunk 1]
```diff
@@ -0,0 +1,257 @@
+{
+    ".source.ts": {
```
Comment (torn4dom4n) [inline snippets/typescript.json]: Add `.source.ts`

File: snippets/typescript.json [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                "import { $0 } from \"${1:module}\";"
+            ],
+            "description": "Import external module."
+        },
+        "Property getter": {
+            "prefix": "get",
+            "body": [
+                "",
+                "public get ${1:value}() : ${2:string} {",
+                "\t${3:return $0}",
```
Comment (torn4dom4n) [inline snippets/typescript.json]: Add `\t`

File: snippets/typescript.json [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+            ],
+            "description": "Log to the console"
+        },
+        "Define a full property": {
+            "prefix": "prop",
+            "body": [
+                "",
+                "private _${1:value} : ${2:string};",
+                "public get ${1:value}() : ${2:string} {",
+                "\treturn this._${1:value};",
```
Comment (torn4dom4n) [inline snippets/typescript.json]: Add `\t`

File: snippets/typescript.json [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+                "}",
+                "public set ${1:value}(v : ${2:string}) {",
+                "\tthis._${1:value} = v;",
```
Comment (torn4dom4n) [inline snippets/typescript.json]: Add `\t`

File: snippets/typescript.json [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+                "return $1;$0"
+            ],
+            "description": "Return statement"
+        },
+        "Property setter": {
+            "prefix": "set",
+            "body": [
+                "",
+                "public set ${1:value}(v : ${2:string}) {",
+                "\tthis.$3 = v;",
```
Comment (torn4dom4n) [inline snippets/typescript.json]: Add `\t`

File: snippets/typescriptreact.json [hunk 6]
```diff
@@ -0,0 +1,260 @@
+{
+    ".source.tsx": {
```
Comment (torn4dom4n) [inline snippets/typescriptreact.json]: Add `.source.tsx`

File: snippets/typescriptreact.json [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+                "import ${1:name} = require('$0');"
+            ],
+            "description": "Import external module."
+        },
+        "Property getter": {
+            "prefix": "get",
+            "body": [
+                "",
+                "public get ${1:value}() : ${2:string} {",
+                "\t${3:return $0}",
```
Comment (torn4dom4n) [inline snippets/typescriptreact.json]: Add `\t`

File: snippets/typescriptreact.json [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+            ],
+            "description": "Log to the console"
+        },
+        "Define a full property": {
+            "prefix": "prop",
+            "body": [
+                "",
+                "private _${1:value} : ${2:string};",
+                "public get ${1:value}() : ${2:string} {",
+                "\treturn this._${1:value};",
```
Comment (torn4dom4n) [inline snippets/typescriptreact.json]: Add `\t`

File: snippets/typescriptreact.json [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+                "}",
+                "public set ${1:value}(v : ${2:string}) {",
+                "\tthis._${1:value} = v;",
```
Comment (torn4dom4n) [inline snippets/typescriptreact.json]: Add `\t`

File: snippets/typescriptreact.json [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+                "return $1;$0"
+            ],
+            "description": "Return statement"
+        },
+        "Property setter": {
+            "prefix": "set",
+            "body": [
+                "",
+                "public set ${1:value}(v : ${2:string}) {",
+                "\tthis.$3 = v;",
```
Comment (torn4dom4n) [inline snippets/typescriptreact.json]: Add `\t`

**Comment by torn4dom4n** [comment]
Typescript extension edited snippets with https://github.com/Microsoft/vscode/commit/f0281ddc2d14d4051c15786464f06390749689a9. It like what I did. I believe my snippets can work and keep up to date with change.
cc/ @damieng

**Comment by damieng** [comment]
I'm just trying to make sure we can stay up to date with the original repo without some knowledge of what conversions are required being lost.

**Pull request merged**